[Bug] Yarn v2 requires bash to be installed

**Describe the bug**

Running yarn v2 in an environment without bash causes a failure with the error message:

`env: can't execute 'bash': No such file or directory`

**To Reproduce**

Run a node app with Yarn v2 in an environment that does not have bash available. (I discovered this while running a CI job with the _node:10.15.1-alpine_ docker image)

The original error is in a private repo but I can create an example project and CI job if desired.


**Environment if relevant (please complete the following information):**

 - OS: Linux
 - Node version: 10.15.1
 - Yarn version: 2.0.0

Minimal repro created: 
https://gitlab.com/probackpacker/minimal-yarn-v2-app/-/jobs/270013509

Pipeline runs without error using Yarn v1. Upgrade to Yarn v2. Pipeline fails. 
The only package that is installed is _axios_.
If the script is [executed with node directly](https://gitlab.com/probackpacker/minimal-yarn-v2-app/-/jobs/270013273) (instead of `yarn run`) the error is 

> Error: Cannot find module 'axios'

instead

--- a/packages/berry-core/sources/scriptUtils.ts
+++ b/packages/berry-core/sources/scriptUtils.ts
@@ -16,7 +16,7 @@ async function makePathWrapper(location: PortablePath, name: Filename, argv0: Na
   if (process.platform === `win32`) {
     await xfs.writeFilePromise(ppath.format({dir: location, name, ext: '.cmd'}), `@"${argv0}" ${args.join(` `)} %*\n`);
   } else {
-    await xfs.writeFilePromise(ppath.join(location, name), `#!/usr/bin/env bash\nexec "${argv0}" ${args.map(arg => `'${arg.replace(/'/g, `'"'"'`)}'`).join(` `)} "$@"\n`);
+    await xfs.writeFilePromise(ppath.join(location, name), `#!/bin/sh\nexec "${argv0}" ${args.map(arg => `'${arg.replace(/'/g, `'"'"'`)}'`).join(` `)} "$@"\n`);
     await xfs.chmodPromise(ppath.join(location, name), 0o755);
   }
 }

